internal/server/middleware.go

```diff
@@ -70,40 +70,31 @@ func handleInfraDestinationHeader(c *gin.Context) error {
 // authenticatedMiddleware is applied to all routes that require authentication.
 // It validates the access key, and updates the lastSeenAt of the user, and
 // possibly also of the destination.
-func authenticatedMiddleware(srv *Server) gin.HandlerFunc {
```

Comment:
I think the name "authenticatedMiddleware" is confusing here where this is just an auth check and scope now, it doesnt actually pass the request through any middleware

Reply:
Very true, I'll rename this to remove "middlware", and I should also update the godoc at the same time.

Reply:
I pushed a commit that renames these to `authenticateRequest`, and `validateRequestOrganization`. What do you think of those names?